src/HotChocolate/Data/src/Data/Projections/Extensions/ProjectionObjectFieldDescriptorExtensions.cs

```diff
@@ -205,10 +206,27 @@ private static FieldMiddleware CreateMiddleware<TEntity>(IProjectionConvention c
                 context = new MiddlewareContextProxy(context, selection, objectType);
             }
 
+            if (context.Operation.Definition.Operation == OperationType.Mutation
```

Comment:
should this be an else if or do we want to run this always?

Reply:
that's a really good question. I think a mutation can return a relay resolver, I'll see what order it makes sense to apply those. That might be a difficult edge case to cover however.

Reply:
looks like an `else if` will work fine here. I pushed some tests with node resolvers in a mutation and it worked just fine.